api/models/Post.js

```diff
@@ -0,0 +1,34 @@
+const db = require ('../dbConfig/init')
+
+module.exports = class Post {
+    constructor(data){
+        this.id = data.id
+        this.title = data.title
+        this.date = data.date
+        this.name = data.name
+        this.body = data.body
+    }
+    static findById (id) {
+        return new Promise (async (resolve, reject) => {
+            try {
+                let postData = await db.query(`SELECT * FROM posts WHERE id = $1;`, [ id ]);
+                let post = new Post(postData.rows[0]);
+                resolve (post);
+            } catch (err) {
+                reject('Post not found');
+            }
+        });
+    }
+
+    static create(title, date, name, body) {
+         return new Promise (async (resolve, reject) => {
```

Comment:
Interesting to see you using Postgres, we used Mongodb and spent ages trying to figure out how to extract the id. So maybe we would have been better off doing what you did